src/main_nep/nep4.cu

```diff
@@ -232,22 +229,11 @@ static void __global__ find_q_scaler(const int N, const float* g_q, float* g_q_s
   }
 }
 
-static __global__ void
-apply_q_scaler(const int N, const NEP4::ANN ann, const float* __restrict__ g_q_scaler, float* g_q)
-{
-  int n1 = threadIdx.x + blockIdx.x * blockDim.x;
-  if (n1 < N) {
-    for (int d = 0; d < ann.dim; ++d) {
-      g_q[n1 + d * N] *= g_q_scaler[d];
-    }
-  }
-}
-
-// Apply the message passing aggregation between neighbors, A*q_theta
 static __global__ void apply_gnn_message_passing(
   const int N,
   const NEP4::Para para,
   const NEP4::ANN ann,
+  const int* __restrict__ g_type,
```

Comment:
I assume these are the atom types?

Reply:
Yes, the atom types (Chemical symbols) are mapped to some integers from 0 to `num_elements` - 1. In this example:
```
C = 0
H = 1
N = 2
```
according to this line in `nep.in`:
```
type 3 C H N
```